### PR TITLE
events: speed up the `newListener` event

### PR DESCRIPTION
**Pull request by Txiaozhe**
```javascript
if (this._events.newListener !== undefined) {
    this.emit('newListener', type, listener);
}
```

### CODE REVIEW

**Comment by legendecas** [comment]
I don't think this is necessary though I have no strong objection to merge it.

Or could you explain what scenario have you encountered so that the checks on if `newListener` is `undefined` is required?

**Comment by Txiaozhe** [comment]
Yes, I will.👌

**Comment by Txiaozhe** [comment]
I have written a bench. But I'm not sure if this is feasible. Would you please check it?

**Comment by legendecas** [comment]
👍 would you like to post your bench results here?

**Comment by Txiaozhe** [comment]
set n = 10, the result is:
events/events-newListener.js n=10: 73,866.15452799527

**Comment by yorkie** [comment]
> set n = 10, the result is:
> events/events-newListener.js n=10: 73,866.15452799527

I think we need this benchmark for before and after versions :)

File: benchmark/events/events-newListener.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  var n = opts.n;
+  
+  emitter.on('newListener', (event, listener) => {});
+
+  bench.start();
+  
+  for (var i = 0; i < n; ++i) {
+    emitter.on('hello', function() {});
+  }
+
```
[the remark this answers is not in the document]
Comment (Txiaozhe) [reply benchmark/events/events-newListener.js]: yes, I will, thank you!

File: benchmark/events/events-newListener.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+var common = require('../common.js');
+var EventEmitter = require('events');
+var emitter = new EventEmitter();
+
+var bench = common.createBenchmark(main, {
+  n: [10],
+});
+
+var handler = function() {};
```
Comment (yorkie) [inline benchmark/events/events-newListener.js]: Use pure declaration `function noop() { }` instead of statement.

**Comment by Txiaozhe** [comment]
awkward ^_^|||

File: benchmark/events/events-newListener.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+var common = require('../common.js');
+var EventEmitter = require('events');
+var emitter = new EventEmitter();
+
+var bench = common.createBenchmark(main, {
+  n: [10]
+});
+
+function handler() {};
```
Comment (yorkie) [inline benchmark/events/events-newListener.js]: Please remove the semi-colon after the function declaration :)

**Comment by Txiaozhe** [comment]
Thank you, I am sorry so match ^__^|||, from this pr, I will know more about your workflow.

**Comment by Txiaozhe** [comment]
Thank you! I will continue to update.

**Pull request closed without merging**